### DISCUSSION TITLE
Elemental Teal in Google Cloud

### DISCUSSION THREAD

**Discussion by mightymouse2045** [rancher/elemental#1355] [General]
Hi,

I cannot work out how to get this working in Google Cloud. I want to deploy this into Google VM's specifically. Is this possible? I have attempted a deployment of the downloaded ISO generated from within Rancher, but creating a custom image of that dies at the last step saying it doesn't contain an OS.

Creating a fully deployed image with TPM, and secure boot locally and exporting is not possible with Google because it's encrypted. The only way I can see at the moment would be to deploy it locally with the tpm emulation enabled, then import that as an image into GCP. However as the docs state you shouldn't use that method in prod.

Are you planning a supported method any time soon?

**Comment by kkaempf** [comment]
Peter, thanks for reaching out.
The ISO failure might be caused by wrong BIOS settings within the Google VM. The ISO required EFI boot to be enabled, it won't work with "legacy" boot setups.

General support of the major cloud providers is on our backlog and you're welcome to help us.

**Reply by mightymouse2045** [reply]
Sorry for late reply got busy - yeah ok I'll try again with my custom image and see if that makes any difference and post the logs in here. 

My main platform is vmware which I can get working, but I'd like to get a teal cluster running in google on VM's which is closer to my customer deployments than just running up clusters in GKE.

**Reply by mightymouse2045** [reply]
I've had no luck with this at all. Tried a couple of other approaches but I can't get past the enabling secure boot option in Google. It doesn't recognise the OS as being capable. This has come from a vmdk with EFI and secure boot enabled, post the OS install stage, shutdown, uploaded and convert from vmdk - but when I go to use that as the source I can't enable secure boot .

I've tried creating an image of the iso as well but it doesn't boot at all.

I'll try creating an existing vm with EFI and secure boot enabled and see if I can swap the disk - but failing that I'm out of ideas
